internal/controller/pkg/revision/deployment.go

```diff
@@ -105,6 +105,9 @@ func buildProviderDeployment(provider *pkgmetav1.Provider, revision v1.PackageRe
 		if cc.Spec.Image != nil {
 			d.Spec.Template.Spec.Containers[0].Image = *cc.Spec.Image
 		}
+		if cc.Spec.Ports != nil {
```

Comment:
It won't make a difference in this case, but would you mind using `len(cc.spec.Ports) > 0` for consistency with other slice types here?